pkg/steps/artifacts.go

```diff
@@ -107,24 +109,47 @@ func (n *TestCaseNotifier) SubTests(prefix string) []*junit.TestCase {
 	if len(names) == 0 {
 		return nil
 	}
+	statuses := make([]coreapi.ContainerStatus, len(pod.Status.ContainerStatuses))
```

Comment:
I think it might be nice for code health to put the digestion of ContainerStatus --> timelines into a separate func and have some unit tests for it